Assets/HoloToolkit/Input/Scripts/Interactions/HandDraggable.cs

```diff
@@ -194,10 +198,19 @@ private void UpdateDragging()
 
             draggingPosition = pivotPosition + (targetDirection * targetDistance);
 
-            if (IsOrientTowardsUser)
+            if (RotationMode == RotationModeEnum.OrientTowardUser)
             {
                 draggingRotation = Quaternion.LookRotation(HostTransform.position - pivotPosition);
             }
+            else if (RotationMode == RotationModeEnum.KeepUpright)
```

Comment:
`KeepUpright` previously was used in conjunction with `OrientTowardUser` so that when the user looks up or down, the object continues to be upright instead of pivoting toward the user. By having these settings be mutually exclusive, it seems like the more common usecase for `KeepUpright` is no longer possible. Additionally, I think that the code to handle uprighting needs to be after the `HostTransform.rotation` is set to its new value, or else `HostTransform.up` won't have the correct value.

I think you will need to have something like this:  

```cs
if (RotationMode == RotationModeEnum.OrientTowardUser || RotationMode == RotationModeEnum.OrientTowardUserAndKeepUpright) 
{
    draggingRotation = Quaternion.LookRotation(HostTransform.position - pivotPosition);
}
else if (RotationMode = RotationModeEnum.LockRotation)
{
    draggingRotation = HostTransform.rotation;
}
else // RotationModeEnum.Default
{
    Vector3 objForward = mainCamera.transform.TransformDirection(objRefForward); // in world space
    Vector3 objUp = mainCamera.transform.TransformDirection(objRefUp);   // in world space
    draggingRotation = Quaternion.LookRotation(objForward, objUp);
}

...

// Apply Final Rotation
HostTransform.rotation = draggingRotation;
if (RotationMode == RotationModeEnum.OrientTowardUserAndKeepUpright)		
{		
    Quaternion upRotation = Quaternion.FromToRotation(HostTransform.up, Vector3.up);		
    HostTransform.rotation = upRotation * HostTransform.rotation;		
}
```